packages/fxa-content-server/app/scripts/views/mixins/third-party-auth-mixin.js

```diff
@@ -206,6 +206,11 @@ export default {
         this.logFlowEvent(`${provider}.signin-complete`);
 
         this.metrics.flush();
+        
+        // Sync service requires a password to be set before it can be used.
+        if (this.relier.isSync()) {
```

Comment:
Should we also check if the account is missing a password?

Reply:
Ah yea we should, updating

Reply:
I tested this a few more times, a user won't be able to do social login for Sync. They are almost prompted to enter password if they have it.